src/vs/workbench/contrib/terminal/browser/xterm/decorationAddon.ts

```diff
@@ -79,6 +81,13 @@ export class DecorationAddon extends Disposable implements ITerminalAddon {
 				this.refreshLayouts();
 			} else if (e.affectsConfiguration('workbench.colorCustomizations')) {
 				this._refreshStyles(true);
+			} else if (e.affectsConfiguration(TerminalSettingId.ShellIntegrationDecorationsEnabled) || e.affectsConfiguration(TerminalSettingId.ShellIntegrationEnabled)) {
+				const enabledDecorations = this._configurationService.getValue(TerminalSettingId.ShellIntegrationDecorationsEnabled) && this._configurationService.getValue(TerminalSettingId.ShellIntegrationEnabled);
+				if (!enabledDecorations) {
+					document.querySelector(DecorationSelector.Container)?.classList.add('hide');
+				} else {
+					document.querySelector(DecorationSelector.Container)?.classList.remove('hide');
+				}
```

Comment:
As mentioned by @LangLangBart in https://github.com/microsoft/vscode/issues/154412#issuecomment-1178702192, we'll want to change the actual decorations to either not exist or change to be invisible (including `overviewRulerOptions`).

Reply:
We could recreate the decorations when re enabled since we will still have the markers. I'm thinking about the extra perf consumed maintaining them if someone has disabled them completely